x/concentrated-liquidity/lp.go

```diff
@@ -613,3 +619,31 @@ func (k Keeper) validatePositionUpdateById(ctx sdk.Context, positionId uint64, u
 
 	return nil
 }
+
+func (k Keeper) settleFinalPoolPosition(ctx sdk.Context, poolId uint64, amount0, amount1 sdk.Int) error {
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	withdrawCoins := sdk.NewCoins(sdk.NewCoin(pool.GetToken0(), amount0.Abs()), sdk.NewCoin(pool.GetToken1(), amount1.Abs()))
+
+	// Check if the amount of liquidity left in the pool is less than the amount to withdraw.
+	amountShort := sdk.NewCoins()
+	for _, coin := range withdrawCoins {
+		poolBalance := k.bankKeeper.GetBalance(ctx, pool.GetAddress(), coin.Denom)
+		if poolBalance.Amount.LT(coin.Amount) {
+			amountShort = amountShort.Add(sdk.NewCoin(coin.Denom, coin.Amount.Sub(poolBalance.Amount)))
+		}
+	}
+
+	// If there is a short amount, move the amount short from the spread rewards address to the pool address.
+	if !amountShort.IsZero() {
```

Comment:
should we add a sanity check, e.g. amountShort.LT(100)

Reply:
So on the discussion Roman and I were having, initially we were just going to send all the dust to the final user anyway.  So this feels fine even if the amount takes up the entire remaining spread rewards account balance

Reply:
I agree with the current approach because we assume that the missing amount is stemming from a rounding error in favor of a fee charge. However, I don't think it is safe to send the entire amount

Reply:
Agreed with having some cap on this amount

Reply:
imo having some cap that keeps this branch in the ballpark of rounding error would be appropriate